src/nimib/capture.nim

```diff
@@ -1,30 +1,40 @@
 ## This submodule implements only one template to temporarily redirect and capture stdout during execution of a chunk of code.
-import tempfile
-
-# see https://stackoverflow.com/questions/64026829/how-to-temporarily-capture-stdout
-# Thanks, Clonk!
-
-# low level, should be Posix only but they happen to work on (my) Windows, too!
-proc dup(oldfd: FileHandle): FileHandle {.importc, header: "unistd.h".}
-proc dup2(oldfd: FileHandle, newfd: FileHandle): cint {.importc, header: "unistd.h".}
+import fusion/ioutils
+import std/os
+import std/tempfiles
 
 template captureStdout*(ident: untyped, body: untyped) =
-  ## redirect stdout to a temporary file and captures output of body in ident
-  let
-    # Duplicate stdout
-    stdoutFileno = stdout.getFileHandle()
-    stdoutDupfd = dup(stdoutFileno)
-    # Create a new temporary file
-    (tmpFile, tmpFilename) = mkstemp(mode=fmWrite)
-    tmpFileFd: FileHandle = tmpFile.getFileHandle()  
-  discard dup2(tmpFileFd, stdoutFileno)  # writing to stdoutFileno now writes to tmpFile
+    captureStdout(ident, "", body)
```

Comment:
This is a template so it might not matter, but I would like you to define this overload of `captureStdout` without `filename` below the other one. I don't know if the CI error is due to this or not.

Reply:
It is possible that the complier is somehow unable to reorder the code when use with Nimib

Reply:
How do you mean? I don't think the compiler does any reordering by default.

Reply:
I also don't really know since template should not need to reorder for it to work.

Reply:
Indeed 👍 I guess it did run without problem for you locally? It could perhaps be something with the other errors that's causing it to give a bad error. So I would reorder templates, add the dependency and commit it and see what the CI says about it

Reply:
Yes, it is working locally and official playground